src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -370,15 +370,61 @@ def deleteBlocks(self):
         """
         _deleteBlocks_
         Find deletable blocks, then decide if to delete based on:
-        Is there an active subscription for dataset or block ?
-          If yes => set deleted=2
-          If no => next check
         Has transfer to all destinations finished ?
-          If yes => request block deletion, approve request, set deleted=1
+          If yes => Delete rules associated with the block, set deleted=1
           If no => do nothing (check again next cycle)
         """
-        # FIXME: figure out the proper logic for rule block deletion
-        logging.info("Starting deleteBlocks methods --> IMPLEMENT-ME!!!")
+        #Get list of blocks that can be deleted
+        blockDict = self.findDeletableBlocks.execute(transaction=False)
+
+        if not blockDict:
+            logging.info("No blocks were found")
+            return
+
+        deletableBlocks = []
+        containerDict = {}
+		#Populate containerDict, assigning each block to its correspondant container
+        for blockName in blockDict:
+            containerList = self.rucio.listParentDIDs(blockName, scope=self.scope)
+            for cont in containerList:
+                if cont['name'] not in containerDict:
+                    containerDict[cont['name']] = {'blocks':[], 'rse':[], 'copies':0}
+                containerDict[cont['name']]['blocks'].append(blockName)
+
+        for contName in containerDict:
+            cont = containerDict[contName]
+            rules = self.rucio.listDataRules(contName, self.scope)
+            #Retrieve requested copies and RSEs
+            for rule in rules:
+                if rule['account'] == self.rucioAcct:
+                    cont['copies'] += rule['copies']
+                    cont['rse'].extend(self.rucio.evaluateRSEExpression(rule['rse_expression'], useCache=True, returnTape=True))
+
+            #Get RSE in which the block is available
+            availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
+
+            for rses in availableRSEs:
+                #Count how many of the copies were requested by this account
+                actualCopies = len(set(rses['replica']) & set(cont['rse']))
+
+                #If enough requested copies are available, block can be deleted
+                if cont['copies'] <= actualCopies:
+                    deletableBlocks.append(rses['name'])
+
+        #Delete agent ccreated rules locking the block
+        binds = []
+        for block in deletableBlocks:
+            rules = self.rucio.listDataRules(block, self.scope)
+            for rule in rules:
+                if rule['account'] == self.rucioAcct:
+                    if self.rucio.deleteRule(rule['id']):
+                        logging.info("Successfully deleted block: %s.", block)
```

Comment:
Please be explicit on these logging records, like
```
logging.info("Successfully deleted rule: %s, for block: %s.", rule, block)
```

same for the logging.error line below (and please, make it a warning level instead of error).

Reply:
I improved logging according to your suggestions